internal/resolve/catalog.go

```diff
@@ -65,6 +66,15 @@ func (r *CatalogResolver) Resolve(ctx context.Context, ext *ocv1.ClusterExtensio
 		}
 	}
 
+	type catStat struct {
```

Comment:
WDYT about extracting this to package-level and making attributes private?

Reply:
I started out with private attributes, but then when these are eventually passed into `l.Info()`, logr only outputs the exported attributes. We could write more code to specifically log each unexported field, but that would be brittle if we ever add new fields.

Not highly opposed to extracting to package level, but I didn't see any use case where this would be used outside the function, so I figured the more local, the better.

Reply:
you're right, though it's actually the `zap` backend sink that controller-runtime/log uses which requires public fields - see https://github.com/go-logr/logr/issues/142

Regarding the declaration, sure it can be extracted in the future if there's a need. Personally I just find it more readable with type declarations done on the outside and functions/methods kept smaller.

Reply:
Hmm, I thought I had switched everything over to the klog backend a month or so ago.

Reply:
I think we should also ensure that the log output is lowercased, which I am not sure happens right now.

With `textlogger` it can be done by making the `catStat` implement `slog.LogValuer` for example, which would also solve the problem of private attributes - ie. we would be free to make them private.

Alternatively, adding tags with lowercase keys to the struct would also work.

Reply:
In the log line, the structured log _keys_ are lowercase. However, the `catStat` value string contains field names that match the casing of the struct fields (hence they are currently capitalized).

I don't personally think we _need_ to downcase the first letter of those field names, but I'm also not really against it. I'll generally agree that the logs would look more consistent if we did.

I'll take a look at what it would take to do that. It might just be json tags.

Reply:
Adding JSON tags fixed it!